lms/static/scripts/frontend_apps/components/dashboard/AllCoursesActivity.tsx

```diff
@@ -51,12 +51,18 @@ export default function AllCoursesActivity() {
     <div className="flex flex-col gap-y-5">
       <h2 className="text-lg text-brand font-semibold">All courses</h2>
       <DashboardActivityFilters
-        selectedStudentIds={studentIds}
-        onStudentsChange={studentIds => updateFilters({ studentIds })}
-        selectedAssignmentIds={assignmentIds}
-        onAssignmentsChange={assignmentIds => updateFilters({ assignmentIds })}
-        selectedCourseIds={courseIds}
-        onCoursesChange={courseIds => updateFilters({ courseIds })}
+        students={{
+          selectedIds: studentIds,
+          onChange: studentIds => updateFilters({ studentIds }),
+        }}
+        assignments={{
+          selectedIds: assignmentIds,
+          onChange: assignmentIds => updateFilters({ assignmentIds }),
+        }}
+        courses={{
+          selectedIds: courseIds,
+          onChange: courseIds => updateFilters({ courseIds }),
+        }}
```

Comment:
I need to check if this causes more renders. In that case we may need to memoize the three objects.